porcupine/plugins/update_check.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+import datetime
+import logging
+
+import requests
+
+from porcupine import __version__ as this_porcupine_version
+from porcupine import settings, utils
+from porcupine.plugins import statusbar
+from porcupine.settings import global_settings
+
+log = logging.getLogger(__name__)
+
+# Place the update checkbox towards the end of the settings dialog
+setup_after = ["restart"]
+
+
+def x_days_ago(days: int) -> str:
+    days_in_year = 365.25  # good enough
+    days_in_month = days_in_year / 12
+    months = round(days / days_in_month)
+
+    if days == 0:
+        return "today"
+    if days == 1:
+        return "yesterday"
+    if days < 1.5 * days_in_month:
+        return f"{days} days ago"
+
+    months = round(days / days_in_month)
+    if months < 12:
+        return f"about {months} months ago"
+    if months == 12:
+        return "about 1 year ago"
+    if months == 13:
+        return "about 1 year and 1 month ago"
+    if months < 24:
+        return f"about 1 year and {months - 12} months ago"
+    if months % 24 == 0:
+        return f"about {months // 12} years ago"
+    if months % 24 == 1:
+        return f"about {months // 12} years and 1 month ago"
+    return f"about {months // 12} years and {months % 12} months ago"
+
+
+def get_date(version: str) -> datetime.date:
+    year, month, day = map(int, version.lstrip("v").split("."))
+    return datetime.date(year, month, day)
+
+
+def fetch_release_creator(version: str) -> str | None:
```

Comment:
Personally I don't see why it would be useful to show, who created a release. It just adds a bunch of complexity to the code.

Reply:
Knowing if it's an automated release vs a an intentional one is nice

Reply:
Porcupine doesn't do automated releases, and probably never will.

Reply:
The GitHub action releases?

Reply:
They are triggered by me or rdbende, not fully automated.